### PR TITLE
19 redis

### LINKED ISSUE #19: Cache county calculated data in Redis
Using Redislabs look into caching the data that is retrieved from AirTable to reduce the number of requests being sent to AirTable

- [x] Push data into Redis
- [x] Pull data from Redis into the backend when a county is requested

### PR DESCRIPTION
**Pull request by jberns**
This closes #19 to cache the calculated data for each county after it is calculated the first time.

An `.env` file will be required to run this branch in order to interact with the redis server.

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -19,10 +37,33 @@ async def counties():
...
-async def get_county_data(county_id: str):
+async def get_county_data(county_id: str) -> CountyResponse:
+    response = CountyResponse(message="Returning county data")
     if county_id in county_list:
         county: str = county_list[county_id]
-        county_data: pd.DataFrame = getData(county)
-        return {"data": county_data}
+
+        county_cache_exists = r.exists(county)
+        if (county_cache_exists == 1 ): # 1 = exists, 0 = does not exist
```
Comment (pjhoberman) [inline backend/main.py:46]: Might be a little less readable, but do you want to do something like

`if r.exists(county): `

File: backend/tests/denver_county_raw.py [hunk 2]
```diff
@@ -0,0 +1 @@
..."% Households below 200% FPL\":0.37,\"EA Participants\":\"3779\",\"LEAP Participants\":\"8874\",\"Total Households Assisted\":12653,\"% of Households below 200% FPL Assisted\":0.12},\"2016\":{\"Population\":663303,\"Households\":281072,\"Average Household Size\":2.31,\"Median Household Income\":56258,\"Households below 200% FPL\":98956,\"% Households below 200% FPL\":0.35,\"EA Participants\":\"3168\",\"LEAP Participants\":\"8313\",\"Total Households Assisted\":11481,\"% of Households below 200% FPL Assisted\":0.12},\"2017\":{\"Population\":678467,\"Households\":287262,\"Average Household Size\":2.31,\"Median Household Income\":60098,\"Households below 200% FPL\":95839,\"% Households below 200% FPL\":0.33,\"EA Participants\":\"3236\",\"LEAP Participants\":\"7432\",\"Total Households Assisted\":10668,\"% of Households below 200% FPL Assisted\":0.11},\"2018\":{\"Population\":693417,\"Households\":294358,\"Average Household Size\":2.31,\"Median Household Income\":63793,\"Households below 200% FPL\":92530,\"% Households below 200% FPL\":0.31,\"EA Participants\":\"3181\",\"LEAP Participants\":\"7160\",\"Total Households Assisted\":10341,\"% of Households below 200% FPL Assisted\":0.11}}"
```
Comment (pjhoberman) [inline backend/tests/denver_county_raw.py:1]: What about combining the three denver data files into one, and then importing from the one file in tests?

File: backend/utils/date_helper.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from datetime import datetime
```
Comment (pjhoberman) [inline backend/utils/date_helper.py:1]: What about combining some of the utils files into one. perhaps `helpers.py` or something could include date helper and redis helper

File: backend/tests/denver_county_raw.py (hunk 2 above)
Comment (jberns) [reply backend/tests/denver_county_raw.py:1]: Updated this.  I think this makes a lot more sense and is way cleaner.

File: backend/utils/date_helper.py (hunk 3 above)
Comment (jberns) [reply backend/utils/date_helper.py:1]: I recombined these for now.  If it gets too big we can split out function later.

**Comment by jberns** [comment]
> Have you seen that?

I think I fixed that testing issue.  I added setup and teardown functions so the redis helper is removed before moving to the next test which fixed this error.

**Pull request merged**